compiler/rustc_borrowck/src/diagnostics/conflict_errors.rs

```diff
@@ -633,10 +632,7 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
             let Some(default_trait) = tcx.get_diagnostic_item(sym::Default) else {
                 return false;
             };
-            // Regions are already solved, so we must use a fresh InferCtxt,
-            // but the type has region variables, so erase those.
-            tcx.infer_ctxt()
-                .build()
+            self.infcx
                 .type_implements_trait(default_trait, [tcx.erase_regions(ty)], param_env)
```

Comment:
we can stop erasing regions now when checking this stuff. Please do this here and at every other location where you've switched to the existing infcx.

```suggestion
                .type_implements_trait(default_trait, [ty], param_env)
```